src/containers/Elements/NodeProvider.js

```diff
@@ -8,61 +8,58 @@ import { actionCreators as networkActions } from '../../ducks/modules/network';
 
 import { NodeList, Node } from '../../components/Elements';
 
+import { diff, nodeIncludesAttributes } from '../../utils/Network';
+
 class NodeProvider extends Component {
   constructor(props) {
     super(props);
-
-    this.state = {
-      nodes: []
-    };
   }
 
-  componentWillReceiveProps(nextProps) {
-    const initalNodeState = { position: {x: 0, y: 0}, isSelected: false };
+  handleSelect = (node) => {
+    // do something with node
+    this.handleAction(node);
+  }
 
-    this.setState({
-      nodes: nextProps.network.nodes.map(() => { return { ...initalNodeState } })
-    });
+  handleStop = (nodeId, node) => {
+    // do something with node
+    this.handleAction(node);
   }
 
-  handleSelect = (node, index) => {
-    const nodes = this.state.nodes;
-    nodes[index].isSelected = !nodes[index].isSelected;
+  handleAction(node) {
+    const {
+      source,
+      activeNodeAttributes,
+      updateNode,
+    } = this.props;
+
+    const updatedNode = { ...node, ...activeNodeAttributes };
 
-    this.setState({ nodes: nodes });
-  }
+    console.log('updateNode', source, node, updatedNode);
 
-  handleStop = (node, index, event, draggableData) => {
-    // const nodes = this.state.nodes;
-    // nodes[index].position = { x: draggableData.x, y: draggableData.y };
-    // this.setState({ nodes: nodes });
-    alert("you moved it, for now we're ignoring that");
+    switch(source) {
```

Comment:
This is going to have two different behaviours, but I agree that this should be removed until that case actually exists.